pkg/fsutil/claim.go

```diff
@@ -53,7 +54,7 @@ func ClaimFile(dir, pattern string) (*os.File, error) {
 	}
 
 	for i := max + 1; ; i++ {
-		name := prefix + strconv.Itoa(i) + suffix
+		name := filepath.Join(dir, prefix+strconv.Itoa(i)+suffix)
```

Comment:
Thanks for fixing the bug, but please do add a unit test.

Reply:
Isn't that your responsibility since I don't think modifying this change to use `ClaimFile` is even necessary or particularly useful? Note that I wouldn't have even noticed this problem if I didn't perform due diligence to see what happens if I launch an Elvish shell built with this change without an already accessible daemon.

Reply:
Well, you could argue that everything is my responsibility and I wouldn't disagree :)

It's good practice to add a regression test when you fix a bug, but if doing that demotivates you from working on this PR I certainly won't insist.